src/agent/coverage/src/block/linux.rs

```diff
@@ -245,13 +251,17 @@ impl Images {
 #[derive(Clone, Debug, PartialEq)]
 pub struct ModuleImage {
     map: MemoryMap,
+    path: ModulePath,
```

Comment:
Why do you wrap `PathBuf` rather than just using `PathBuf` directly?

Reply:
Its ctor validates some additional conditions on a `PathBuf`, which lets us treat otherwise-partial and fallible accessors as total. It also documents the provenance of the path, and lets us keep our resulting assumptions in one place.